engines/sci/graphics/video32.cpp

```diff
@@ -0,0 +1,393 @@
+	_boostEndColor = CLIP<int16>(boostEndColor, 0, 255);
+	_leaveScreenBlack = flags & kPlayFlagLeaveScreenBlack;
+	_leaveLastFrame = flags & kPlayFlagLeaveLastFrame;
+	_blackPalette = flags & kPlayFlagBlackPalette;
+	_stretchVertical = flags & kPlayFlagStretchVertical;
+}
+
+VMDPlayer::IOStatus VMDPlayer::close() {
+	if (!_isOpen) {
+		return kIOSuccess;
+	}
+
+	_decoder->close();
+	_isOpen = false;
+	_isInitialized = false;
+
+	if (!_planeIsOwned && _screenItem != nullptr) {
+		g_sci->_gfxFrameout->deleteScreenItem(*_screenItem);
+		_screenItem = nullptr;
+	} else if (_plane != nullptr) {
+		g_sci->_gfxFrameout->deletePlane(*_plane);
+		_plane = nullptr;
+	}
+
+	if (!_leaveLastFrame && _leaveScreenBlack) {
+		// This call *actually* deletes the plane/screen item
+		g_sci->_gfxFrameout->frameOut(true);
+	}
+
+	if (_blackoutPlane != nullptr) {
+		g_sci->_gfxFrameout->deletePlane(*_blackoutPlane);
+		_blackoutPlane = nullptr;
+	}
+
+	if (!_leaveLastFrame && !_leaveScreenBlack) {
+		// This call *actually* deletes the blackout plane
+		g_sci->_gfxFrameout->frameOut(true);
+	}
+
+	if (!_showCursor) {
+		g_sci->_gfxCursor->kernelShow();
+	}
+
+	_lastYieldedFrameNo = 0;
+	_planeIsOwned = true;
+	_priority = 0;
+	return kIOSuccess;
+}
+
+VMDPlayer::EventFlags VMDPlayer::kernelPlayUntilEvent(const EventFlags flags, const int16 lastFrameNo, const int16 yieldInterval) {
+	assert(lastFrameNo >= -1);
+
+	const int32 maxFrameNo = (int32)(_decoder->getFrameCount() - 1);
+
+	if ((flags & kEventFlagToFrame) && lastFrameNo > 0) {
+		_decoder->setEndFrame(MIN((int32)lastFrameNo, maxFrameNo));
+	} else {
+		_decoder->setEndFrame(maxFrameNo);
+	}
+
+	if (flags & kEventFlagYieldToVM) {
+		_yieldInterval = 3;
+		if (yieldInterval == -1 && !(flags & kEventFlagToFrame)) {
+			_yieldInterval = lastFrameNo;
+		} else if (yieldInterval != -1) {
+			_yieldInterval = MIN((int32)yieldInterval, maxFrameNo);
+		}
+	} else {
+		_yieldInterval = maxFrameNo;
+	}
+
+	return playUntilEvent(flags);
+}
+
+VMDPlayer::EventFlags VMDPlayer::playUntilEvent(const EventFlags flags) {
+	// Flushing all the keyboard and mouse events out of the event manager to
+	// avoid letting any events queued from before the video started from
+	// accidentally activating an event callback
+	for (;;) {
+		const SciEvent event = _eventMan->getSciEvent(SCI_EVENT_KEYBOARD | SCI_EVENT_MOUSE_PRESS | SCI_EVENT_MOUSE_RELEASE | SCI_EVENT_QUIT);
+		if (event.type == SCI_EVENT_NONE) {
+			break;
+		} else if (event.type == SCI_EVENT_QUIT) {
+			return kEventFlagEnd;
+		}
+	}
+
+	_decoder->pauseVideo(false);
+
+	if (flags & kEventFlagReverse) {
+		// NOTE: This flag may not work properly since SCI does not care
+		// if a video has audio, but the VMD decoder does, for some reason.
+		assert(_decoder->setReverse(true));
+		_decoder->setVolume(0);
+	}
+
+	if (!_isInitialized) {
+		_isInitialized = true;
+
+		if (!_showCursor) {
+			g_sci->_gfxCursor->kernelHide();
+		}
+
+		Common::Rect vmdRect(
```

Comment:
And speaking of the bikesheding (pardon?), we had a lot of discussions in the past, and it is very important to have common code formatting in a 1.8M+ SLOC codebase. Otherwise it quickly will turn into an unmaintainable mess.